### ISSUE TITLE
calcite-button: Angular two-way data binding broken in Chrome and Safari

### ISSUE THREAD

**Issue by andygup** [Esri/calcite-design-system#7729]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

In the JS SDK Editor widget, calcite-buttons that require two-way data binding to the button's properties are no longer updating in Angular 16 applications. This is a regression from `@esri/calcite-components@1.2.0`. In our current use case, buttons are not correctly being enabled or disabled.

Note, the functionality works in Firefox 116, however it is not working in Chrome 116 or Safari 16.

### Expected Behavior

Two-way data binding should work on all Calcite components being used with JS SDK widgets in an Angular application.

### Reproduction Sample

https://devtopia.esri.com/andy4683/angular-editor-update

### Reproduction Steps

1. Clone the [repo](https://devtopia.esri.com/andy4683/angular-editor-update)
2. Follow the instructions in the README. The app is already built with the unminified API, so it is ready for immediate testing.
3. To test a working version ensure:
    - `@arcgis/core` is to set to version `4.27.0-next.20230426`
    - `@esri/calcite-components` is set to version `1.2.0`
4. Run `npm i` → `npm run build` → `ng serve` and open the local instance in any browser
5. Select the "Select" button under the Editor widget's "Edit features" section.
6. Select an item in the map
7. In the editor widget, change any of the properties and try to select the "Update" button
8. Observe the behavior (functional)
9. To test a non-working version ensure:
    - `@arcgis/core` is to set to version `4.27.0-next.20230426`
    - `@esri/calcite-components` is set to version `1.2.1-next.0`
10. Repeat steps 4-7 above
11. In particular, observe the behavior in a Chromium browser (not functional) and compare to FF (functional).

### Reproduction Version

`1.2.1-next.0`

### Relevant Info

Verified using Angular 16 and 17-beta.

This issue showed up at `@arcgis/core@4.27.0-next.20230426` when using `@esri/calcite-compoments@1.2.1-next.0`. The last known working combination was `@arcgis/core@4.27.0-next.20230426` using `@esri/calcite-compoments@1.2.0`.

[Diff between 1.2.0 and 1.2.1-next.0](https://github.com/Esri/calcite-design-system/compare/v1.2.0...v1.2.1-next.0)

### Regression?

Yes, @esri/calcite-components@1.2.0 was the last working version.

### Priority impact

p1 - need for current milestone

### Impact

This potentially affects all users, business partners and distributors using the latest version of @arcgis/core with Angular.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by geospatialem** [comment]
A few updates:
- The bug is present in `1.2.1-next.0`+ but not `1.2.0` with `@arcgis/core` version `4.27.0-next.20230426`. Here's the [diff between the two versions](https://github.com/Esri/calcite-design-system/compare/v1.2.0...v1.2.1-next.0)
- This seems to only effect Chromium browsers (FF is unaffected at this time)
- Angular 16 was released on May 3 this year
- `calcite-button`'s properties seem to be respected, with the exception of `disabled`, which could be a culprit in this instance

**Comment by geospatialem** [comment]
Reassigning to November - likely dependent on the Angular wrapper issue #7860.

**Comment by andygup** [comment]
Another update:
- This also affecting ArcGIS Components (beta). I was able to reproduce the issue using [`arcgis-editor`](https://developers.arcgis.com/javascript/latest/components/storybook/?path=/docs/map-components_component-reference-editor--docs) in Angular 16.

**Comment by benelan** [comment]
Hi @andygup! Do you know if `arcgis-editor` uses Stencil's Angular output target? 

We recently added the output target to our monorepo and published it as [`@esri/calcite-components-angular`](https://www.npmjs.com/package/@esri/calcite-components-angular). A starter app using `v1.11.0-next.0` is available [here](https://github.com/Esri/calcite-components-examples/tree/main/angular), which uses Angular v16 internally.

There is also a starter app in the [previous commit](https://github.com/Esri/calcite-components-examples/tree/f6b8b070cf40e20b25a638ae967c102adc6629b7/angular) using the release candidate, which uses Angular 17 and Stencil 4 internally.

> Note: The current release candidate (`v2.0.0-rc.0`) requires installing `@esri/calcite-components` separately, which will not be the case moving forward.

Do you mind re-testing this issue in one or both of the starter apps mentioned above to see if it is resolved? I'll do some testing on my end as well, thanks!

**Comment by andygup** [comment]
> Do you know if arcgis-editor uses Stencil's Angular output target?

Hi @benelan! I'm not sure on this one.

> @esri/calcite-components-angular

Awesome. I'll test on Angular 16 and 17. Btw, we are doing some extensive with Angular 17.  When you get a minute, take a look at JS SDK issue 7022. I noticed your sample is defaulting to webpack instead of esbuild, which is our recommendation at this time.

**Comment by andygup** [comment]
@benelan I did two initial tests on the Angular 16 [sample](https://github.com/Esri/calcite-components-examples/tree/main/angular):
- [x] basic Angular data binding in `calcite-button` **does** appear to be working in Angular 16. I was able to programmatically enable and disable the button using both the Angular pattern (e.g. `[disabled]="disabled"`) and standard JavaScript pattern where you use `getElementById` and toggle `disabled`.
- [ ] tested the Editor widget and the bug is **still present** 😦 . I created a local distribution build of @arcgis/core using CC `1.11.0-next.2`.  I'll double check my work tomorrow.

----------

Packages used for testing:

```json
    "@arcgis/core": "file:arcgis-core-4.29.0-next.20231115.tgz",
    "@esri/calcite-components-angular": "1.11.0-next.0",
    "@esri/calcite-components": "~1.11.0-next.2", <== used to build @arcgis/core
```
Here's dependency graph for @arcgis/core:

<img width="478" alt="image" src="https://github.com/Esri/calcite-design-system/assets/510440/92ff6749-8d41-435f-99f0-da6f94bceb78">

**Comment by andygup** [comment]
@benelan I'm trying to build the Angular 17 sample using `@esri/calcite-components-angular@2.0.0-rc.0`, and I'm getting a bunch of TypeScript errors?

<img width="897" alt="image" src="https://github.com/Esri/calcite-design-system/assets/510440/ded89d4d-af75-43b6-961d-66cb25867d07">

**Comment by benelan** [comment]
> @benelan I'm trying to build the Angular 17 sample using `@esri/calcite-components-angular@2.0.0-rc.0`, and I'm getting a bunch of TypeScript errors?

Can you use `2.0.0-rc.1`? `rc.0` had an issue due to Angular's strange packaging workflow, see #8151 for context.


> tested the Editor widget and the bug is still present 

This sounds related related to the arcgis web components, I can take a look.

**Comment by benelan** [comment]
> Can you use 2.0.0-rc.1? rc.0 had an issue due to Angular's strange packaging workflow, see https://github.com/Esri/calcite-design-system/pull/8151 for context.

Hmm never mind, `rc.0` works fine for me. It's probably related to this note from one of my previous comments

> Note: The current release candidate (`v2.0.0-rc.0`) requires installing `@esri/calcite-components` separately, which will not be the case moving forward.

However `rc.1` doesn't have that requirement, so I recommend testing with that version anyway.

**Comment by andygup** [comment]
> This sounds related to the arcgis web components, I can take a look.

For the Editor widget, I was importing it directly from @arcgis/core.

> However rc.1 doesn't have that requirement, so I recommend testing with that version anyway.

I'm still seeing the same TypeScript errors using `2.0.0-rc.1`

**Comment by andygup** [comment]
UPDATE: this is starting to look like a dependency issue.

When using the wrapper `rc0` or `rc1`, the TypeScript errors are showing up only on Angular 16 and not Angular 17.

**Comment by benelan** [comment]
Gotcha, we bumped Angular to 17 on our release candidate branch, so it's probably related to that. 

I was just about a working sample on the [`benelan/angular-rc`](https://github.com/Esri/calcite-components-examples/tree/benelan/angular-rc) branch of the examples repo, but looks like you figured it out.

**Comment by andygup** [comment]
> Gotcha, we bumped Angular to 17 on our release candidate branch

Okay, agreed that would explain it. I'll proceed with testing `rc1` and the Editor widget against Angular 17.

**Comment by andygup** [comment]
Blocked - I'll retest when we have an `@arcgis/core@next` branch that's using CC v2

**Comment by geospatialem** [comment]
Adding the `blocked` label per https://github.com/Esri/calcite-design-system/issues/7729#issuecomment-1821185777 for additional Maps SDK testing - reallocated to the December milestone.

**Comment by andygup** [comment]
Unblocked, tested - issue is still valid with CC v2 `rc.1` and `rc.2`. /cc @geospatialem 

**TL;DR** Tested with 4.28/4.29-next, CC `2.0.0-rc.1`, Calcite Angular wrapper `2.0.0-rc.1` and Angular 17. All testing results point to the use of `VNodeChild` with `calcite-button`. Any other widget using this pattern will have the same issue. Other widgets that are using a standard calcite-button as an `HTMLElement` will most likely work okay. 

**Data-binding is working with CC.** Angular data binding seems to be working correctly with standalone calcite-button, as well as with stand-alone calcite-button in other widgets such as Directions. And, it's working correctly with other Calcite components such as calcite-slider.

**Workaround.** The current workaround is to disable zone.js monkey patching for click events. This may break some customers apps.

**Future.** hopefully customers building new Angular apps at Angular 17+ will disable zone.js. But at least for Angular 17, zone.js is still enabled by default.

**Comment by andygup** [comment]
Interesting note - Compass does not use `VNodeChild`, it's just using `calcite-button` by itself. Not sure if that completely negates my `VNodeChild` theory.

**Comment by andygup** [comment]
Update for 4.29: known widgets affected include zoom, compass and Editor.

**Comment by maxpatiiuk** [comment]
I found out the cause of the issue. This is an Angular bug and it has been present there for 5 years. See https://github.com/angular/angular/issues/54581

See https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/59883#issuecomment-4603240 and https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/59883#issuecomment-4603272 for why this bug affects Calcite when used with Maquette

**Comment by andygup** [comment]
@geospatialem this issue is documented in the Angular sample on [jsapi-resources](https://github.com/Esri/jsapi-resources/tree/main/esm-samples/jsapi-angular-cli#known-issues) as a Known Issue. That text will be updated with the new information/clarification about the Angular bug on Monday when 4.29 releases.

Seems like we can close this issue out as a known limit. What are your thoughts?

**Comment by geospatialem** [comment]
> Seems like we can close this issue out as a known limit. What are your thoughts?

@andygup Agreed, this is outside the scope of Calcite. Closing as unplanned for now. cc @benelan 

Thanks for the deep dive and providing the findings, @maxpatiiuk! 💪🏻

**Comment by andygup** [comment]
Quick update - the Angular bug has been fixed in zone.js 0.14.5: https://github.com/angular/angular/compare/zone.js-0.14.4...zone.js-0.14.5
